packages/api/src/controllers/passwordResetController.js

```diff
@@ -0,0 +1,140 @@
+        const diffDays = Math.abs(sendEmailDate - created_at) / (1000 * 60 * 60 * 24);
+        if (diffDays > 1) {
+          reset_token_version = 1;
+          created_at = sendEmailDate;
+        } else if (reset_token_version === 3) {
+          return res.status(400).send('Reached maximum number of available reset tokens');
+        } else {
+          reset_token_version++;
+        }
+
+        // generate token
+        // payload: user_id, reset_token_version, email, first_name
+
+        const updateData = {
+          reset_token_version,
+          created_at: created_at.toISOString(),
+        };
+
+        const pwResult = await passwordModel.query().findById(userData.user_id).update(updateData);
+
+        const tokenPayload = {
+          ...userData,
+          email,
+          reset_token_version: reset_token_version - 1,
+          created_at: created_at.getTime(),
+        };
+        const token = await createResetPasswordToken(tokenPayload);
+
+
+        const template_path = '../templates/password_reset_email.html';
+        const subject = 'LiteFarm Farm Access Restored Email';
+        const replacements = {
+          first_name: userData.first_name,
+        };
+        const sender = 'system@litefarm.org';
+        await emailSender.sendEmail(template_path, subject, replacements, email, sender, `/callback/?reset_token=${token}`);
+
+        return res.status(200).send('Email successfully sent');
+      } catch (error) {
+        console.log(error);
+        return res.status(400).json(error);
+      }
+    };
+  }
+
+  static validateToken() {
+    return async (req, res) => {
+      // passwordResetController.isTokenValid()
+      return res.status(200).json({ isValid: true });
+    };
+  }
+
+  static resetPassword() {
+    return async (req, res) => {
+      // if(passwordResetController.isTokenValid())
+      // reset the password
+      // set create_at to today on the password table
+      // send email
+      // log the user in
+      const { password } = req.body;
+      const { user_id, email, first_name } = req.user;
+      try {
+        // hash password
+        const salt = await bcrypt.genSalt(10);
+        const password_hash = await bcrypt.hash(password, salt);
+
+        const pwData = {
+          password_hash,
+          reset_token_version: 0,
+          created_at: new Date().toISOString(),
+        };
+        const pwResult = await passwordModel.query().findById(user_id).update(pwData);
+
+        // generate token for logging user in
+        const id_token = await createAccessToken({ user_id });
+
+        const template_path = '../templates/reset_password_confirmation.html';
+        const subject = 'Your LiteFarm password has been changed';
+        const replacements = {
+          first_name,
+        };
+        const sender = 'system@litefarm.org';
+        await emailSender.sendEmail(template_path, subject, replacements, email, sender, `/?email=${email}`);
```

Comment:
There is a flaw with the URL that is being sent in the email. If users use a `+` symbol in email for their LiteFarm account, the email will not be displayed correctly when the field in automatically populated.